website/docs/wallet/withdraw-validator.md

```diff
@@ -0,0 +1,404 @@
+  -H 'accept: application/json'
+```
+
+where you can see its value in the response, where it says “withdrawal_credentials”:
+
+```rust
+{
+  "execution_optimistic": false,
+  "data": {
+    "index": "1",
+    "balance": "1",
+    "status": "active_ongoing",
+    "validator": {
+      "pubkey": "0x93247f2209abcacf57b75a51dafae777f9dd38bc7053d1af526f220a7489a6d3a2753e5f3e8b1cfe39b56f43611df74a",
+      "withdrawal_credentials": "0xcf8e0d4e9587369b2301d0790347320302cc0943d5a1884560367e8208d920f2",
+      "effective_balance": "1",
+      "slashed": false,
+      "activation_eligibility_epoch": "1",
+      "activation_epoch": "1",
+      "exit_epoch": "1",
+      "withdrawable_epoch": "1"
+    }
+  }
+}
+```
+
+### Step 3: **Sign your BLS to Execution change**
+
+Now that the staking deposit tool is executable, you can then use it to generate your signed **[BLS to Execution](https://github.com/ethereum/staking-deposit-cli/blob/bls-to-execution-change/README.md#generate-bls-to-execution-change-arguments)** request. You need to use your mnemonic for this step, so doing it offline is key and ensuring you do not paste your mnemonic anywhere else than necessary.
+
+:::caution
+⚠️ We recommend doing this next step *without* an Internet connection to be maximally secure. Either turn off the internet before introducing your mnemonic for signing or migrate to an air-gapped environment to continue the following steps.
+:::
+
+Here’s the command to get started with the process. This command will **not** submit your signed message to the network yet, and will only generate the data needed for the next steps.
+
+```jsx
+python ./staking_deposit/deposit.py generate-bls-to-execution-change
+```
+
+By calling the command above, you should go through an interactive process that will ask you for the following information:
+
+1. **Your mnemonic language**. You can see the different options available, where English is one of the options, among others
+2. **The network** you wish to perform this operation for. Select the `zhejiang` network for this tutorial
+3. Enter your **mnemonic** next
+4. Next, you will be asked for your the index of the withdrawal key you wish to use. For **most users**, this will be the default of 0 unless you know what you are doing.
+5. You will then be asked the **validator** **indices** for the validators you wish to generate the message for. You can find your validator indices on block explorers such as [https://beaconcha.in](https://beaconcha.in) or in your Prysm validator client logs. For example, the validator with public key `0x8078c7f4ab6f9eaaf59332b745be8834434af4ab3c741899abcff93563544d2e5a89acf2bec1eda2535610f253f73ee6` on [https://beacocha.in](https://beacocha.in) has validator index 8 by navigating to its [page](https://beaconcha.in/validator/8)
+6. Next you will be asked for your **withdrawal credentials,** which you should now have if you followed this guide
```

Comment:
Why does the staking cli require the withdrawal credentials by default? This should be computed by default. If this is the case we should raise an issue with @hwwhww 

In fact this guide says that the withdrawal credentials should be needed, but I claim the normal workflow should not need them, just is a nice-to-have to check later that the from-bls-pubkey field is actually correct, before submitting the message.

Reply:
One user might have used the same mnemonics to generate multiple deposits. For mistake-proofing, `staking-deposit-cli` checks if the input old BLS withdrawal credentials are identical to the computed BLS withdrawal credentials. If they mismatch, the cli throws error messages immediately.